### PR TITLE
initial GHA workflows for "build test" and "remote push"

### PR DESCRIPTION
**Pull request by andy-neuma**
SUMMARY:
* initial set of "actions with a little a" that are the building blocks for eventual CI system
* "build test" workflow
* "remote push" workflow on `a10g`
* update some requirement files to have packages listed in alphabetical order

NOTE: this PR is still somewhat nebulas as i'm still working through building and testing "neuralmagic-vllm" in our automation environment. 

TEST:
currently, i'm working through various workflow components, i.e. "actions with a little a". the bits making up the actions in this PR have been constructed from my notes along the way.

we can do a "complete" run that includes: linting, building, installing, and running tests.

GHA link ... https://github.com/neuralmagic/neuralmagic-vllm/actions/runs/7975058564
`testmo` ... https://neuralmagic.testmo.net/automation/runs/view/8097

Latest GHA link ... https://github.com/neuralmagic/neuralmagic-vllm/actions/runs/7992489982

### CODE REVIEW

**Comment by varun-sundar-rabindranath** [comment]
Hey Andy - perhaps it makes sense to rename all files to have a `neuralmagic` prefix ? (I think it'd be cumbersome - but will about merge conflicts with upstream)

File: .github/actions/nm-test-vllm/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: test neuralmagic-vllm
+description: "test neuralmagic-vllm via, 'pytest tests/'"
+inputs:
+  test_directory:
+    description: 'test directory, path is relative to magic_wand'
```
Comment (varun-sundar-rabindranath) [inline .github/actions/nm-test-vllm/action.yml:5]: nit : cruft `reference to magic_wand`

**Review by varun-sundar-rabindranath — APPROVED** [review]

**Comment by varun-sundar-rabindranath** [comment]
Looks good in general - had a couple of comments, but nothing major and could be iterated on !

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # formatting
-yapf==0.32.0
```
[the remark this answers is not in the document]
Comment (andy-neuma) [reply requirements-dev.txt:2]: i'm okay with it.

File: .github/actions/nm-test-vllm/action.yml (hunk 1 above)
Comment (andy-neuma) [reply .github/actions/nm-test-vllm/action.yml:5]: oh, good catch. :)

File: .github/actions/nm-build-vllm/action.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: build neuralmagic-vllm
+description: 'build neuralmagic-vllm'
+inputs:
+  Gi_per_thread:
+    description: 'requested Gi to reserve per thread'
```
Comment (tlrmchlsmth) [inline .github/actions/nm-build-vllm/action.yml:5]: is this GiB? Could you clarify?

File: .github/scripts/determine-threading [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -e
+
+usage() {
+    echo "Usage: ${0} <options>"
+    echo
+    echo "  -G    - number of Gi per processor (includes hyperthreads, default is 1 Gi)."
```
Comment (tlrmchlsmth) [inline .github/scripts/determine-threading:6]: ditto, Gi -> GiB?

**Review by tlrmchlsmth — APPROVED** [review]

**Pull request merged**